packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactDelegate.java

```diff
@@ -237,17 +238,29 @@ public boolean onKeyLongPress(int keyCode) {
 
   public void reload() {
     DevSupportManager devSupportManager = getDevSupportManager();
-    if (devSupportManager != null) {
-      // With Bridgeless enabled, reload in RELEASE mode
-      if (devSupportManager instanceof ReleaseDevSupportManager
-          && ReactFeatureFlags.enableBridgelessArchitecture
-          && mReactHost != null) {
-        // Do not reload the bundle from JS as there is no bundler running in release mode.
-        mReactHost.reload("ReactDelegate.reload()");
+    if (devSupportManager == null) {
+      return;
+    }
+
+    // Reload in RELEASE mode
+    if (devSupportManager instanceof ReleaseDevSupportManager) {
+      // Do not reload the bundle from JS as there is no bundler running in release mode.
+      if (ReactFeatureFlags.enableBridgelessArchitecture) {
+        if (mReactHost != null) {
+          mReactHost.reload("ReactDelegate.reload()");
+        }
       } else {
-        devSupportManager.handleReloadJS();
+        UiThreadUtil.runOnUiThread(() -> {
+          if (mReactNativeHost.hasInstance() && mReactNativeHost.getReactInstanceManager() != null) {
+            mReactNativeHost.getReactInstanceManager().recreateReactContextInBackground();
+          }
+        });
```

Comment:
Why can't we implement this in `ReleaseDevSupportManager`?

Reply:
sounds good. there's an only question about feature parity for bridgeless support. could we implement the reloading feature for bridgeless inside `ReleaseDevSupportManager`? or should we introduce another `BridgelessReleaseDevSupportManager`?

Reply:
if we only have `ReleaseDevSupportManager`, is it okay to contain `mReactHost.reload()` inside the class? given the current ReleaseDevSupportManager doesn't contain any ReactHost code.

or do you mean to keep bridgeless `mReactHost.reload()` the current `ReactDelegate.reload()`. and only move bridge `mReactNativeHost.getReactInstanceManager().recreateReactContextInBackground()` into ReleaseDevSupportManager?

Reply:
You are actually right. `ReleaseDevSupportManager` doesn't contain ReactHost code which is why I ended up adding the check for `instanceOf ReleaseDevSupportManager` instead in https://github.com/facebook/react-native/pull/43645/

For first approach to get rid of check and move the implementation to `ReleaseDevSupportManager`, you could do the following:

1. Add a new constructor for `ReleaseDevSupportManager`, and pass the `ReactHost` here:
https://github.com/facebook/react-native/blob/995d5b832d3ce354b908f1d8b138bda824260f91/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/ReactHostImpl.java#L187

2. Retain the ReactHost instance as nullable inside the ReleaseDevSupportManager.

3. Implement `ReleaseDevSupportManager.handleReloadJS` and just call `mReactHost.reload()`

You should not need to reload the bundle from JS as there is no bundler running in release mode. Is that what you meant by moving the implementation to `ReleaseDevSupportManager` @javache ?

However, the issue with Bridge in RELEASE mode would still need to be addressed.

Reply:
> You should not need to reload the bundle from JS as there is no bundler running in release mode.

This may still be a valid use-case, if you want to reset the entire application state for example.

We could either implement bridgeless support in `ReleaseDevSupportManager` or add a new `BridgelessReleaseDevSupportManager`.

Reply:
> We could either implement bridgeless support in ReleaseDevSupportManager or add a new BridgelessReleaseDevSupportManager.

i can go with either direction. it would involve some refactoring, e.g. adding an extra parameter to the constructor: `public ReleaseDevSupportManager() {` -> `public ReleaseDevSupportManager(ReactHost reactHost) {`.

what if i would like to the have reloading feature on react-native 0.74. just to double confirm, does the refactoring make sense to happen in the period?

Reply:
The other problem with having `public ReleaseDevSupportManager(ReactHost reactHost)` is the introduction of circular dependency of `devsupport` -> `runtime` -> `devsupport`

https://github.com/facebook/react-native/blob/b7de91666453b832b889d9577226097aeeace1a8/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/runtime/BridgelessDevSupportManager.java#L23-L26